### PR TITLE
optimize UX for loading pipeline pages

### PR DESCRIPTION
**Pull request by IronPan**
speed up the query time for list runs

For example:
Before 
```
SELECT subq.uuid, 
       "[" 
       || Group_concat(r.payload, ",") 
       || "]" AS refs 
FROM   (SELECT rd.*, 
               "[" 
               || Group_concat(m.payload, ",") 
               || "]" AS metrics 
        FROM   (SELECT * 
                FROM   run_details 
                WHERE  uuid IN (SELECT resourceuuid 
                                FROM   resource_references AS rf 
                                WHERE  ( rf.resourcetype = 'Run' 
                                         AND rf.referenceuuid = 'b7cc0970-e823-431a-a7d9-6b34407bbe17' 
                                         AND rf.referencetype = 'Experiment' ))) AS rd 
               LEFT JOIN run_metrics AS m 
                      ON rd.uuid = m.runuuid 
        GROUP  BY rd.uuid) AS subq 
       LEFT JOIN (SELECT * 
                  FROM   resource_references 
                  WHERE  resourcetype = 'Run') AS r 
              ON subq.uuid = r.resourceuuid 
GROUP  BY subq.uuid 
ORDER  BY createdatinsec ASC, 
          uuid ASC 
LIMIT  40;
```

After
```
SELECT subq.uuid, 
       "[" 
       || Group_concat(r.payload, ",") 
       || "]" AS refs 
FROM   (SELECT rd.*, 
               "[" 
               || Group_concat(m.payload, ",") 
               || "]" AS metrics 
        FROM   (SELECT * 
                FROM   run_details 
                WHERE  uuid IN (SELECT resourceuuid 
                                FROM   resource_references AS rf 
                                WHERE  ( rf.resourcetype = 'Run' 
                                         AND rf.referenceuuid = 'b7cc0970-e823-431a-a7d9-6b34407bbe17' 
                                         AND rf.referencetype = 'Experiment' )) ORDER  BY createdatinsec ASC, uuid ASC LIMIT  40) AS rd 
               LEFT JOIN run_metrics AS m 
                      ON rd.uuid = m.runuuid 
        GROUP  BY rd.uuid) AS subq 
       LEFT JOIN (SELECT * 
                  FROM   resource_references 
                  WHERE  resourcetype = 'Run') AS r 
              ON subq.uuid = r.resourceuuid 
GROUP  BY subq.uuid; 
ORDER  BY createdatinsec ASC, uuid ASC;
```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1085)

### CODE REVIEW

**Comment by IronPan** [comment]
/hold
more optimization might be needed for further speedup

File: backend/api/run.proto [hunk 1]
```diff
@@ -150,7 +150,8 @@ message TerminateRunRequest {
 }
 
 message ListRunsResponse {
-  repeated Run runs = 1;
+  repeated Run runs = 1 [deprecated=true];
```
[the remark this answers is not in the document]
Comment (IronPan) [reply backend/api/run.proto]: reverted.

File: backend/src/apiserver/storage/run_store.go [hunk 2]
```diff
@@ -145,21 +145,18 @@ func (s *RunStore) buildSelectRunsQuery(selectCount bool, opts *list.Options,
 	}
 
 	sqlBuilder := opts.AddFilterToSelect(filteredSelectBuilder)
-	if err != nil {
```
Comment (hongye-sun) [inline backend/src/apiserver/storage/run_store.go]: Why is this removed?

File: backend/src/apiserver/storage/job_store.go [hunk 3]
```diff
@@ -112,15 +112,13 @@ func (s *JobStore) buildSelectJobsQuery(selectCount bool, opts *list.Options,
...
-		return "", nil, util.NewInternalServerError(err, "Failed to list jobs: %v", err)
-	}
 
 	// If we're not just counting, then also add select columns and perform a left join
 	// to get resource reference information. Also add pagination.
 	if !selectCount {
+		sqlBuilder = opts.AddPaginationToSelect(filteredSelectBuilder)
 		sqlBuilder = s.addResourceReferences(sqlBuilder)
-		sqlBuilder = opts.AddPaginationToSelect(sqlBuilder)
+		sqlBuilder = opts.AddSortingToSelect(sqlBuilder)
```
Comment (hongye-sun) [inline backend/src/apiserver/storage/job_store.go]: Just wonder if it's still required to add sorting query if inner result has already been sorted.
